lib/match.js

```diff
@@ -63,11 +63,8 @@ function compareArrays(array, expected, partialMode) {
  */
 function compareObjects(object, expected, partialMode) {
   if (
-    !compareArrays(
-      Object.keys(object).sort(),
-      Object.keys(expected).sort(),
-      partialMode
-    )
+    !partialMode &&
+    Object.keys(object).length !== Object.keys(expected).length
   )
```

Comment:
Comparing the number of the keys should be sufficient here since, below we compare the content.